.azure-pipelines/steps/run-in-docker.yml

```
@@ -42,6 +42,9 @@ steps:
         --env IncludeMinorPackageVersions=$(IncludeMinorPackageVersions) \
         --env NUGET_ENABLE_EXPERIMENTAL_HTTP_RETRY=true \
         --env enable_crash_dumps=$(enable_crash_dumps) \
+        --env DD_LOGGER_API_KEY=$(DD_LOGGER_API_KEY) \
+        --env DD_LOGGER_SERVICE=$(DD_LOGGER_SERVICE) \
+        --env DD_LOGGER_ENV=$(DD_LOGGER_ENV) \
```

If you include the explicit set, then this fails if you _haven't_ set the values.
If you leave them blank, they're included if available, and not otherwise. That _may_ be a way to skirt around the issues you're seeing, just don't include them where you don't want them?

```suggestion
        --env DD_LOGGER_API_KEY \
        --env DD_LOGGER_SERVICE \
        --env DD_LOGGER_ENV \
```

---

well, those values are always specified, they are global variables in the ultimate pipeline yaml.